### PR TITLE
Balance: Remove Bactrian from all non-mission Pirate fleets

### PR DESCRIPTION
**Pull request by alexrovw**
**Balance**

As discussed on Discord, this change will balance acquisition of a Bactrian by removing default hostile Bactrians from non-story-related fleets.

## Summary

Specifically, we are removing all NPC Pirate gov Bactrian fleet variants from:
* The Pirate fleets that roam around the North
* The Hired Gun Bactrian fleets from Hai Reveal part 1
* The friendly escort Bactrian fleet during the positive-rep pirate job: `mission "Escort Stolen Vessel (Extra Large, North, Dangerous Path)"`

I'd love some feedback on that last one.

The reasoning for this change is two-fold:

1. Gameplay: There is an existing mission string that grants _purchase rights_ to the Bactrian. This mission reward is greatly devalued if the player has access to free Pirate Bactrians just by wandering around the North or by triggering a Pirate job.
2. Lore: The Deep tightly controls Bactrians, to the point where you cannot buy them without a license. It's pretty unlikely that they would let _so many_ Pirate Bactrians roam around the North. It makes sense for one-off warlords like Beelzebub to have a Bactrian as their prized ship, but it's just not all that believable that random Pirates are flying Bactrians around.

The player can still capture an NPC Bactrian if they really want to, but they have to pay a heavier price. The remaining NPC Bactrians are all Merchants, so pirating them will result in a heavy reputation hit. We are still able to showcase the Bactrian in action, because NPC Merchant fleets will still have them. Finally, Beelzebub still has a nicely outfitted Bactrian (the "Cosmic Devil") that the player can earn if they work for it at the conclusion of the mission string.

### CODE REVIEW

**Comment by Hecter94** [comment]
I think this is a good initiative, and I like the direction you're going with this, but we could go a bit further and accomplish your stated goals more effectively. 

I believe a better approach is leaving Bactrians in pirate fleets and allowing capturing them. However, if the player lands on a Deep or Republic world with a Bactrian without having the requisite license, they are approached by Deep Security and strong-armed into selling the Bactrian to them for a generous finder's fee or else suffer a large reputation penalty and likely turn them hostile by refusing to hand the ship over to them. 

This gives us the opportunity for a fun, short mission where the player can choose to acquiesce and give up their prize for a still decent reward, perhaps try to convince the Deep government otherwise (potentially acting as an alternative entry route into the Bactrian mini-campaign?), or defy the government and accept the consequences for doing so. 

This effectively accomplishes the same goal of increasing the value of the mission chain that will award the license, as it will now enable both purchasing and effectively capturing the Bactrian. 
In addition, it reinforces the lore point of Bactrians being tightly controlled by the Deep government in a way that I would argue is more engaging than simply removing them outright, which also further reinforces the lore point of the Republic's reliance on licenses, which has for a while been a lore point that hasn't been effectively shown via gameplay or story.

This alternate solution also maintains a greater degree of player freedom. It encourages creative thinking, as you're still able to capture a Bactrian but need to be careful with how you use it. For example, capturing a Bactrian, bringing it straight to Free Worlds space, and using it exclusively there without landing on any Republic or Deep-controlled planets would become a valid strategy. Once the pirate campaign is further fleshed out, you would have some interesting alternatives for pirate-minded players to capture Bactrians for themselves and use them exclusively at the fringes of society where they won't be persecuted for their actions. 

In summary, while I don't disagree with the reasoning for this change in theory, we have better options for accomplishing the stated goals more thoroughly while simultaneously reducing the impact on player freedom. However, We could still go with this as an interim solution if we feel the change is urgent.

**Comment by Amazinite** [comment]
> It's pretty unlikely that they would let *so many* Pirate Bactrians roam around the North.

This seems rather hyperbolic to me to italicize "so many." The Bactrian will only spawn in 1 out of every 64 large northern pirate fleets. The best spawn rate I can find for large northern pirate fleets is a spawn period of 500 frames in Arneb. 500 frames per fleet * 64 fleets before you see a Bactrian = about 9 minutes of idling in a pirate fleet to see a pirate Bactrian. (3 minutes on fast forward.) If you're anywhere else, that time between seeing pirate Bactrians will only increase. We could easily increase that time if we wanted, but it's already one of if not the rarest ship to find in a pirate fleet.

I'd be fine with removing the Bactrian from the job and Hired Gun fleets (although with the job I'd prefer to replace it with a different fleet variant rather than removing a fleet variant entirely), but I don't think it makes sense to entirely eliminate the possibility of seeing a Bactrian fleet spawn in large pirate fleets.

> Lore: The Deep tightly controls Bactrians, to the point where you cannot buy them without a license.

Let's make a comparison to the Navy. Navy ships are license locked and don't appear in pirate fleets. The claim is effectively that since the Bactrian is also license locked, it should similarly not appear in pirate fleets.

I think the issue with this is that in-universe, there's a difference in the exclusivity of the Navy and city-ship licenses. In order to gain a Navy license, one must join the Navy. By joining the Navy, you'll go through a training program to enhance your combat capabilities. It's noted at multiple points in the story, such as the Alpha attack of Poisonwood and during the Wanderer campaign with Avalon and Zenith, that the Navy are highly adept at H2H combat. The rest of your crew will also be a part of the Navy, all with the same combat training. This means that for a pirate to even capture a Navy ship in the first place, they have to go toe-to-toe with the combat experienced crew of the ship, making the odds of a pirate capturing a Navy ship lower. And even in the likely rare cases where it does occur, the Navy is a powerful force in the galaxy. Their odds of success in destroying or returning lost ships will be very high.

By contrast, the City-Ship license is not as exclusive. It isn't only given to Deep Security, but also to merchants who have a good reputation with the Deep. Although it is a larger ship with more crew, merchants are less experienced in H2H combat than the Navy or Deep Security (who one can imagine get similar H2H combat training to the Navy). Deep Security is also a relatively less powerful force than the Navy. They only patrol a small region of space and are lesser in numbers. While they do have the means to travel out to other regions of space, and can likely ask the Navy for help when they need it, it's imaginable that their success rate with destroying or returning lost ships isn't as high as the Navy.

Another point is that the Deep mention during FWR that they only have legal jurisdiction to track down fugitives under certain circumstances. From "FW Reconciliation 2A":
> "When we joined the Republic, we retained the authority to police our own territory, and that authority includes the right to track down fugitives who have fled beyond our usual jurisdiction."
> "Our proposal is that a fleet under our jurisdiction, but joined by Free Worlds and Oathkeeper vessels, will travel into the Core and arrest at least one of the Syndicate masterminds behind the terrorist attacks. Such an action would satisfy the technical requirements of legality provided that immediately prior to the arrest, one of our courts hears Sawyer's testimony and pronounces that Syndicate individual guilty."`

It's notable that in order to legally justify the arrest, they need to have an open case in one of the Deep's courts. This may not be possible with all stolen Bactrians.

TL;DR I'm fine with removing Bactrian spawns from some of these locations, but don't think they should be removed as the rarest fleet spawn of large northern pirate fleets.

**Comment by alexrovw** [comment]
> This seems rather hyperbolic to me to italicize "so many."

Perhaps that is a bit of an exaggeration. But anecdotally, I've been able to capture 3-4 pirate Bactrians in a typical playthrough without actively looking for them. Spending enough time in the northern systems inevitably leads to Bactrian encounters. Considering the fact that these are old and rare ships as well as the fact that they are controlled by the Deep, I would expect that finding even one in the hands of pirates would be a rarity.

> By contrast, the City-Ship license is not as exclusive. It isn't only given to Deep Security, but also to merchants who have a good reputation with the Deep.

I would think that the City-Ship license would be more exclusive. Surely not everyone in DeepSec is allowed to fly a Bactrian, while practically everyone in the Navy either flies or serves aboard a Navy ship. There are also fewer Bactrians by far when compared to Navy ships.

**Comment by Amazinite** [comment]
> Perhaps that is a bit of an exaggeration. But anecdotally, I've been able to capture 3-4 pirate Bactrians in a typical playthrough without actively looking for them... I would expect that finding even one in the hands of pirates would be a rarity.

How long is a "typical playthrough" for you? A 1/64 chance for an event that only occurs occasionally in most locations is rare relative to other ship appearances in the game, but given enough time and/or luck, even the rarest events can happen multiple times. Although like I said, that can easily be increased in its rarity.

**Comment by alexrovw** [comment]
Consider that there are 36 systems that have a chance to spawn `Large Northern Pirates` (though not at the same rate) as well as numerous missions and jobs that can generate them too. My typical playthrough is long, but I think anyone who makes a bunch of trips to Hai/Wanderer space and back is bound to run into them reasonably often.

**Comment by Amazinite** [comment]
> My typical playthrough is long

Well unless we make them almost impossible to spawn, you're still bound to see some on a long playthrough. What do you imagine an acceptable spawn rate to be for the length of your playthroughs?

**Comment by alexrovw** [comment]
To be honest, I was hoping to remove them from pirate fleets entirely, outside of scripted mission encounters. But if we're just looking to decrease the spawn rate, I'd say cutting it in half would be a welcome change.

**Comment by Amazinite** [comment]
I'll note too that I'm not sold on the idea of making getting a Bactrian impossible outside of getting the license for the sake of making the mission getting the license better. Sure, the license is a reward for getting to the end of the Deep missions, but missions shouldn't simply be boiled down to the material rewards that you get from them. The journey itself should be the most important part.

**Comment by Amazinite** [comment]
> But if we're just looking to decrease the spawn rate, I'd say cutting it in half would be a welcome change.

In that case, you can double the `variant #` values in the large northern pirate fleet for everything aside from the Bactrian's variant.

**Comment by alexrovw** [comment]
> In that case, you can double the `variant #` values in the large northern pirate fleet for everything aside from the Bactrian's variant.

Done

> The journey itself should be the most important part.

Agreed. But when somebody is going into the mission for the express purpose of earning the Bactrian, they typically complain about it being too long. Then, at the end, they don't even get a Bactrian – only the right to purchase one. This PR isn't intended to fully address that concern, but it's supposed to make the Bactrian a bit more exclusive for those who want to earn one.
